spec/index.md

```diff
@@ -2340,6 +2340,30 @@ This function returns fee percentiles, measured in millisatoshi/vbyte (1000 mill
 
 The [standard nearest-rank estimation method](https://en.wikipedia.org/wiki/Percentile#The_nearest-rank_method), inclusive, with the addition of a 0th percentile is used. Concretely, for any i from 1 to 100, the ith percentile is the fee with rank `⌈i * 100⌉`. The 0th percentile is defined as the smallest fee (excluding coinbase transactions).
 
+### IC method `fetch_logs` {#ic-fetch_logs}
+
+Provides the logs of the canister including trap messages.
+Logs are stored in a circular buffer of 4KiB.
+Log memory persists across upgrades and it is deleted if the canister is reinstalled or uninstalled.
+The log visibility is defined in `log_visibility` field of `canister_settings`.
+Users cannot call this method.
+
+### IC method `fetch_logs_query` {#ic-fetch_logs_query}
+
+This method is identical to [`fetch_logs`](#ic-fetch_logs), but exposed as a query.
+
+:::note
+
+This query is only accessible in non-replicated mode. Calls in replicated mode are rejected.
+
+:::
```

Comment:
Can you explain the rationale behind this? Why wouldn't it be ok to run this in replicated mode instead of duplicating functionality in an update call?

Reply:
which speaks against having the `fetch_logs` method defined above as an update call whatsoever

Reply:
I think we can remove the replication version of `fetch_logs` and only start with the non-replicated version. We can always introduce the replicated mode later.

Reply:
> We can always introduce the replicated version later.

i.e., do not reject replicated calls, introducing a separate method sounds weird;

I'd also drop the suffix _query

Reply:
do you mean I should drop the replicated endpoint `fetch_logs` and rename the non-replicated by dropping the `_query` suffix?

what if we later decide to add a replicated endpoint, would it be possible to have two endpoints with the same name?

Reply:
The same endpoint can be used for replicated and non-replicated executions, just like regular canister queries.

Reply:
Long-term, what do we envision in terms of charging for this endpoint (e.g., once we have charging for queries in place)? Would it be free, or charged to the canister?

Reply:
> do you mean I should drop the replicated endpoint fetch_logs and rename the non-replicated by dropping the _query suffix?

yes

Reply:
Done.